miden-tx/tests/swap_script_test.rs

```diff
@@ -107,7 +108,82 @@ fn test_swap_script() {
 
     let requested_note = OutputNote::new(recipient, note_assets, note_metadata);
 
-    let created_note = transaction_result.output_notes().get_note(0);
+    let created_note = executed_transaction.output_notes().get_note(0);
 
     assert_eq!(created_note, &requested_note);
 }
+
+#[test]
+fn test_swap_prove_and_verify() {
```

Comment:
Is there a benefit to adding a separate test like this? In `P2IDR` test, we just integrated proving/verification into the existing test. Should we do the same here?

Reply:
agreed.